mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/proxy/MQTTProxyProtocolMethodProcessor.java

```diff
@@ -107,12 +109,9 @@ public void processConnect(Channel channel, MqttConnectMessage msg) {
         } else {
             MQTTAuthenticationService.AuthenticationResult authResult = authenticationService.authenticate(payload);
             if (authResult.isFailed()) {
-                MqttMessage connAck = MqttConnAckMessageHelper.
-                        createMqtt(
-                                MqttUtils.isMqtt5(protocolVersion)
-                                        ? MqttConnectReturnCode.CONNECTION_REFUSED_BAD_USERNAME_OR_PASSWORD :
-                                        MqttConnectReturnCode.CONNECTION_REFUSED_BAD_USER_NAME_OR_PASSWORD
-                        );
+                MqttMessage connAck = MqttUtils.isMqtt5(protocolVersion)
+                        ? MqttConnAckMessageHelper.createMqtt(Mqtt5ConnReasonCode.BAD_USERNAME_OR_PASSWORD) :
+                        MqttConnAckMessageHelper.createMqtt(Mqtt3ConnReasonCode.CONNECTION_REFUSED_NOT_AUTHORIZED);
```

Comment:
Seems we should use Mqtt3ConnReasonCode.BAD_USERNAME_OR_PASSWORD which missed 0x04.

Reply:
fixed :)